aws/src/main/java/nl/nn/adapterframework/filesystem/AmazonS3FileSystem.java

```diff
@@ -92,6 +96,8 @@ public class AmazonS3FileSystem extends FileSystemBase<S3Object> implements IWri
 	private @Getter Integer proxyPort = null;
 	private @Getter int maxConnections = 50;
 
+	private @Getter String storageClass = null;
```

Comment:
If the storageClass applies to every operation on every file, it would make more sense to put it on the bucket?

If you want to apply a specific storage class to files after specific operations, should we then not look at how to add storage class as an option / parameter to an operation (such as the "move file" operation)?

Then storage classes can be defined on a more granular level.

Reply:
> If the storageClass applies to every operation on every file, it would make more sense to put it on the bucket?

This is true for this specific instance of the AmazonS3FileSystem that sets the storage class, there could be other instances (in other adapters for example) where the storage class is not set but operate on the same bucket. 

For example in my use case, I'd upload the files under the default storage class but move them to glacier storage once a certain process has been completed. 

I am also not aware on how to configure another storage class as the default on a bucket. A quick google suggests to use lifecycle rules to transition objects between storage classes, but why would you pay money to AWS to do this for you if you could set the desired storage class during creation of the object :)

Besides that, I totally agree that being able to set the desired storage class trough a parameter would enable the Frank developer to use the same instance with different storage classes, which definitely will have its related use-cases.